### PR TITLE
CR-2915 - uninstall

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by vfarcic — COMMENTED** [review]
Are the tests coming later? No TDD?

File: cmd/install/install.go [hunk 1]
```diff
@@ -215,7 +216,7 @@ func waitForDeployments(ctx context.Context, opts *options) {
 	ns := values.Namespace
 	o := &kube.WaitOptions{
 		Interval: time.Second * 2,
-		Timeout:  time.Minute * 2,
+		Timeout:  time.Minute * 20,
```
Comment (roi-codefresh) [inline cmd/install/install.go]: this is insane, lets make it 5min

File: cmd/install/install.go [hunk 2]
```diff
@@ -327,7 +305,7 @@ func persistGitopsRepo(ctx context.Context, opts *options) {
 	cferrors.CheckErr(err)
 
 	if !hasRemotes {
```
Comment (roi-codefresh) [inline cmd/install/install.go]: `isNewRepo`

File: cmd/install/install.go [hunk 3]
```diff
@@ -356,6 +334,15 @@ func initializeNewGitopsRepo(ctx context.Context, opts *options) {
 	values.GitopsRepoClonePath = values.TemplateRepoClonePath
 	values.GitopsRepo, err = git.Init(ctx, values.GitopsRepoClonePath)
 	cferrors.CheckErr(err)
+
+	conf, err := envman.LoadConfig(values.GitopsRepoClonePath)
+	cferrors.CheckErr(err)
+
+	env := conf.FirstEnv()
+	env.TemplateRef = opts.baseRepo
```
Comment (roi-codefresh) [inline cmd/install/install.go]: env.UpdateTemplateRef()

File: cmd/uninstall/uninstall.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	RepoName  string
+	GitToken  string
+}
+
+func New(ctx context.Context) *cobra.Command {
+	var opts options
+
+	cmd := &cobra.Command{
+		Use:   "uninstall",
+		Short: "uninstalls an Argo Enterprise solution from a specified cluster and installation",
```
Comment (roi-codefresh) [inline cmd/uninstall/uninstall.go]: capitalize

File: cmd/uninstall/uninstall.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+		Long:  "this command will clear all Argo-CD managed resources relating to a specific installation, from a specific cluster",
```
Comment (roi-codefresh) [inline cmd/uninstall/uninstall.go]: capitalize

File: cmd/install/install.go [hunk 6]
```diff
@@ -155,10 +151,10 @@ func prepareBase(ctx context.Context, opts *options) {
...
 	log.G(ctx).Debug("renaming envName files")
-	cferrors.CheckErr(helpers.RenameFilesWithEnvName(ctx, values.TemplateRepoClonePath, values.EnvName))
+	cferrors.CheckErr(helpers.RenameFilesWithEnvName(ctx, values.TemplateRepoClonePath, opts.envName))
 
 	log.G(ctx).Debug("rendering template values")
-	cferrors.CheckErr(helpers.RenderDirRecurse(filepath.Join(values.TemplateRepoClonePath, "**/*.*"), values))
+	cferrors.CheckErr(helpers.RenderDirRecurse(filepath.Join(values.TemplateRepoClonePath, "**/*.*"), renderValues))
 }
 
 func tryCloneExistingRepo(ctx context.Context, opts *options) {
```
Comment (roi-codefresh) [inline cmd/install/install.go]: use cloneExisting

File: pkg/git/git.go [hunk 7]
```diff
@@ -134,6 +137,36 @@ func getRef(cloneURL string) string {
 	return u.Fragment
 }
 
+func CloneExistingRepo(ctx context.Context, repoOwner, repoName, gitToken string) (Repository, error) {
```
Comment (roi-codefresh) [inline pkg/git/git.go]: move this to provider interface, get GetRepo options instead of args

Comment (roi-codefresh) [reply pkg/git/git.go]: and use this in install cmd

File: pkg/git/git.go [hunk 8]
```diff
@@ -134,6 +137,36 @@ func getRef(cloneURL string) string {
...
+	p, err := New(&Options{
```
Comment (roi-codefresh) [inline pkg/git/git.go]: NewProvider

File: pkg/git/git.go [hunk 9]
```diff
@@ -282,6 +315,15 @@ func (r *repo) HasRemotes() (bool, error) {
 	return len(remotes) > 0, nil
 }
 
+func (r *repo) Root() (string, error) {
```
Comment (roi-codefresh) [inline pkg/git/git.go:285]: look in install flow where we use values.Repo value

File: pkg/kube/kube.go [hunk 10]
```diff
@@ -51,6 +55,11 @@ type (
 		// DryRunStrategy by default false, can be set to either "client" or "server" dry-run modes, see kubectl apply --help
 		DryRun bool
 	}
+
+	DeleteOptions struct {
+		// IOStreams the std streams used by the apply command
```
Comment (roi-codefresh) [inline pkg/kube/kube.go:60]: add dryrun options

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**